### DISCUSSION TITLE
Internationalization

### DISCUSSION THREAD

**Discussion by ylavoie** [ledgersmb/LedgerSMB#6543] [Design]
Our current strategy for internationalization of LedgerSMB is based onto Locale::MakeText and Transifex.
The latter allows a group of translators to set the translations for many languages which are then proposed to the user by the former.
This strategy has served us well but it has some annoying limitations:
- While it does good for languages with singular and plural, tt is difficult to handle languages with many plural forms, like Russian (3) or Arabic (6)
- Word order is handled in the code and not in the translations, so developers have to be aware that it may change depending on the user language selected.

We are in search of a solution for those issues which will allows the language to be user dependent and put the translation differences outside of code, back in the Transifex interface.

Some possible solutions may be:
- Locale::TextDomain::OO for the Perl portion
- di18n/localize for the Dojo

**Comment by ehuelsmann** [comment]
@ylavoie even with locale maketext, word order should be determined in the translation. I don't think that particular bit makes this a good case. Are there other properties in the other libraries (not in maketext) that do make a good case for this ticket?
